OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -72,6 +72,11 @@ public class ViewportControllerWidget : Widget
 		ScrollDirection edgeDirections;
 		World world;
 		WorldRenderer worldRenderer;
+		WPos viewPortBookmarkSlot1;
```

Comment:
You could cut out a lot of duplication by making this an array of WPos and then extracting helper methods for `SaveBookmark(int index, WPos position)` and `JumpToBookmark(int index)`